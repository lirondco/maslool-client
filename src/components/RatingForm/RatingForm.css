.RatingForm {
    margin-bottom: 2%;
    padding: 1%;
    border: #524C57 solid;
    border-radius: 10px;
    box-shadow: 5px 5px 2.5px grey;
    text-align: center;
    margin-bottom: 1%;
    font-weight: bolder;
}