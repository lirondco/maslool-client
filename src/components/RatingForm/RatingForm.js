import React, { Component } from "react";
import StarRatings from "react-star-ratings";
import TrailContext from "../../contexts/TrailContext";
import TokenService from "../../services/token-service";
import TrailsApiService from "../../services/trails-api-service";
import "./RatingForm.css";

export default class RatingForm extends Component {
  static contextType = TrailContext;

  addRating = (newRating) => {
    const { trail } = this.context;
    TrailsApiService.postRating(trail.id, newRating)
      .then(this.context.addRating)
      .catch(this.context.setError);
  };

  getRatingId = () => {
    const jwtPayload = TokenService.parseAuthToken();
    const { ratings } = this.context;
    const arrayRatings = [...ratings];
    let filteredRating = arrayRatings.filter(
      (rating) => rating.user.id === jwtPayload.user_id
    );
    if (filteredRating.length === 0) {
      return null;
    } else {
      return filteredRating[0].id;
    }
  };

  getRating = () => {
    const jwtPayload = TokenService.parseAuthToken();
    const { ratings } = this.context;
    const arrayRatings = [...ratings];
    let filteredRating = arrayRatings.filter(
      (rating) => rating.user.id === jwtPayload.user_id
    );
    if (filteredRating.length === 0) {
      return null;
    } else {
      return filteredRating[0].rating;
    }
  };

  editRating = (newRating) => {
    const ratingId = this.getRatingId();
    TrailsApiService.editRating(ratingId, newRating)
    .then(() => this.context.editRating(ratingId, {rating: newRating}))
    .catch(this.context.setError);
  };

  render() {
    const ratingId = this.getRatingId();
    const rating = this.getRating();
    return (
      <div className="RatingForm">
        <p>Your Rating:</p>
        {ratingId ? (
          <StarRatings
            rating={rating}
            changeRating={this.editRating}
            numberOfStars={5}
            starDimension="30px"
            starSpacing="15px"
            starRatedColor="red"
            starEmptyColor="grey"
            name="rating"
          />
        ) : (
          <StarRatings
            rating={0}
            changeRating={this.addRating}
            numberOfStars={5}
            starDimension="30px"
            starSpacing="15px"
            starRatedColor="red"
            starEmptyColor="grey"
            name="rating"
          />
        )}
      </div>
    );
  }
}
