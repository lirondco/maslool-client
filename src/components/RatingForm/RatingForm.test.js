import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { withContext } from "shallow-with-context";
import RatingForm from "./RatingForm";

describe(`RatingForm component`, () => {
    const context = {
        ratings: [],
        trail: { id: 999 },
      };
      const ComponentWithContext = withContext(RatingForm, context);
  it("renders the complete form", () => {
    const wrapper = shallow(<ComponentWithContext />, { context });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("validates form on submit", () => {
    const wrapper = shallow(
      <ComponentWithContext />,
      { context }
    );

    

    const mockedEvent = {
      preventDefault: () => {},
      target: { rating: { value: 4 } }
    };
    wrapper.find("StarRatings").simulate("change", mockedEvent);
  });
});
