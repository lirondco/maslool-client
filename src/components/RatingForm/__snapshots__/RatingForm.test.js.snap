// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RatingForm component renders the complete form 1`] = `
<div
  className="RatingForm"
>
  <p>
    Your Rating:
  </p>
  <StarRatings
    changeRating={[Function]}
    gradientPathName=""
    ignoreInlineStyles={false}
    name="rating"
    numberOfStars={5}
    rating={0}
    starDimension="30px"
    starEmptyColor="grey"
    starHoverColor="rgb(230, 67, 47)"
    starRatedColor="red"
    starSpacing="15px"
    svgIconPath="m25,1 6,17h18l-14,11 5,17-15-10-15,10 5-17-14-11h18z"
    svgIconViewBox="0 0 51 48"
    typeOfWidget="Star"
  />
</div>
`;
